src/hotspot/share/opto/compile.cpp

```diff
@@ -1453,7 +1453,8 @@ const TypePtr *Compile::flatten_alias_type( const TypePtr *tj ) const {
     case Type::RawPtr:   tj = TypeRawPtr::BOTTOM;   break;
     case Type::AryPtr:   // do not distinguish arrays at all
     case Type::InstPtr:  tj = TypeInstPtr::BOTTOM;  break;
-    case Type::KlassPtr: tj = TypeKlassPtr::OBJECT; break;
+    case Type::AryKlassPtr:
```

Comment:
Similar situation here. Why don't you leave `Type::KlassPtr` intact and just extend it to `Type::AryKlassPtr` and `Type::InstKlassPtr` cases?